molecule/provisioners/libvirtprovisioner.py

```diff
@@ -0,0 +1,923 @@
+        ET.SubElement(backing, 'format', type='qcow2')
+        volxml = ET.tostring(vol)
+        LOG.debug("\tXMLDesc: {}".format(volxml))
+        newvol = pool.createXML(volxml)
+        utilities.print_success("\tCreated volume for {}.\n".format(instance[
+            'name']))
+
+    def _destroy_volume(self, pool, instance):
+        """
+        Destroy a libvirt storage volume for an instance
+
+        :param: Libvirt Storage Pool object
+        :param: a dict that describes an instance
+        """
+        utilities.print_info(
+            "\t\tDestroying libvirt volume for instance {}".format(instance[
+                'name']))
+        try:
+            vol = pool.storageVolLookupByName(instance['name'] + '.img')
+            vol.delete()
+        except libvirt.libvirtError:
+            LOG.warning("\t\tNo volume for {}".format(instance['name']))
+            return
+        utilities.print_success('\t\tDestroyed libvirt volume for {}'.format(
+            instance['name']))
+
+    def _populate_image(self, instance):
+        """
+        Get this instance's image config.
+
+        Any config specified for this instance is merged with (overriding) any config that already exists for an image of the same name.
+
+        :param: a dict of an instance
+        :return: the final image dict for that instance
+        """
+        image = instance.get(
+            'image', self.molecule.config.config['libvirt']['images'][0])
+        for k in ['source', 'ssh_user', 'ssh_key']:
+            image[k] = image.get(k, filter(
+                lambda imgname: imgname['name'] == image['name'],
+                self.molecule.config.config['libvirt']['images'])[0][k])
+        for index, inst in enumerate(self.molecule.config.config['libvirt'][
+                'instances']):
+            if inst['name'] == instance['name']:
+                if 'image' in self.molecule.config.config['libvirt'][
+                        'instances'][index]:
+                    self.molecule.config.config['libvirt']['instances'][index][
+                        'image'].update(image)
+                else:
+                    self.molecule.config.config['libvirt']['instances'][index][
+                        'image'] = image
+        return image
+
+    def up(self, no_provision=True):
+        """
+        Up or define/up libvirt instances.
+
+        :param: no_provision is not meaningful for libvirt instances
+        """
+        for net in self.molecule.config.config['libvirt']['networks']:
+            self._up_network(net)
+        pool = self._up_pool()
+        vols = pool.listAllVolumes()
+        domains = self._libvirt.listAllDomains()
+        wait = True
+        for instance in self.instances:
+            image = self._populate_image(instance)
+            imagefile = image['source'].split('/')[-1]
+            # Ensure that the image is available
+            if not os.path.exists(
```

Comment:
This isn't really readable because of the new line happening in the middle of the condition.  Please assign the call to `os.path.exists()` to a variable and use the variable in the test. (x2)